auth/auth.go

```diff
@@ -222,20 +219,51 @@ func (auth *Authenticator) calculateHistory(invalSeq uint64, invalGrants ch.Time
 			currentHistoryForGrant = GrantHistory{}
 		}
 
-		// TODO: Will perform pruning here once full
-
 		// Add grant to history
-		currentHistoryForGrant.UpdatedAt = time.Now().UnixNano()
+		currentHistoryForGrant.UpdatedAt = time.Now().Unix()
 		currentHistoryForGrant.Entries = append(currentHistoryForGrant.Entries, GrantHistorySequencePair{
 			StartSeq: previousInfo.Sequence,
 			EndSeq:   invalSeq,
 		})
 		currentHistory[previousName] = currentHistoryForGrant
 	}
 
+	if prunedHistory := currentHistory.PruneHistory(auth.ClientPartitionWindow); len(prunedHistory) > 0 {
+		base.Debugf(base.KeyCRUD, "rebuildChannels: Pruned principal history on %s for %s", base.UD(princName), base.UD(strings.Join(prunedHistory, ",")))
```

Comment:
Would be better to redact each element individually, so each channel name gets a hash, which is already supported for string slices.

```suggestion
		base.Debugf(base.KeyCRUD, "rebuildChannels: Pruned principal history on %s for %s", base.UD(princName), base.UD(prunedHistory))
```